code/game/objects/structures/mortar.dm

```diff
@@ -9,109 +9,104 @@
-		if(do_after(user, 30, TRUE, src, BUSY_ICON_GENERIC))
-			user.visible_message("<span class='notice'>[user] finishes adjusting [src]'s firing angle and distance.</span>",
-			"<span class='notice'>You finish adjusting [src]'s firing angle and distance to match the new coordinates.</span>")
-			busy = 0
-			targ_x = temp_targ_x
-			targ_y = temp_targ_y
-			var/offset_x_max = round(abs((targ_x + dial_x) - x)/offset_per_turfs) //Offset of mortar shot, grows by 1 every 10 tiles travelled
-			var/offset_y_max = round(abs((targ_y + dial_y) - y)/offset_per_turfs)
-			offset_x = rand(-offset_x_max, offset_x_max)
-			offset_y = rand(-offset_y_max, offset_y_max)
-		else busy = 0
-	if(choice == "Dial")
-		var/temp_dial_x = input("Set longitude adjustement from -10 to 10.") as num
-		if(temp_dial_x + targ_x > world.maxx || temp_dial_x + targ_x < 0)
-			to_chat(user, "<span class='warning'>You cannot dial to this coordinate, it is outside of the area of operations.</span>")
-			return
-		if(temp_dial_x < -10 || temp_dial_x > 10)
-			to_chat(user, "<span class='warning'>You cannot dial to this coordinate, it is too far away. You need to set [src] up instead.</span>")
-			return
-		var/temp_dial_y = input("Set latitude adjustement from -10 to 10.") as num
-		if(temp_dial_y + targ_y > world.maxy || temp_dial_y + targ_y < 0)
-			to_chat(user, "<span class='warning'>You cannot dial to this coordinate, it is outside of the area of operations.</span>")
-			return
-		var/turf/T = locate(targ_x + temp_dial_x, targ_y + temp_dial_y, z)
-		if(get_dist(loc, T) < 10)
-			to_chat(user, "<span class='warning'>You cannot dial to this coordinate, it is too close to your mortar.</span>")
-			return
-		if(temp_dial_y < -10 || temp_dial_y > 10)
-			to_chat(user, "<span class='warning'>You cannot dial to this coordinate, it is too far away. You need to set [src] up instead.</span>")
-			return
-		if(busy)
-			to_chat(user, "<span class='warning'>Someone else is currently using this mortar.</span>")
-			return
-		user.visible_message("<span class='notice'>[user] starts dialing [src]'s firing angle and distance.</span>",
-		"<span class='notice'>You start dialing [src]'s firing angle and distance to match the new coordinates.</span>")
-		busy = 1
-		playsound(loc, 'sound/items/ratchet.ogg', 25, 1)
-		if(do_after(user, 15, TRUE, src, BUSY_ICON_GENERIC))
-			user.visible_message("<span class='notice'>[user] finishes dialing [src]'s firing angle and distance.</span>",
-			"<span class='notice'>You finish dialing [src]'s firing angle and distance to match the new coordinates.</span>")
-			busy = 0
-			dial_x = temp_dial_x
-			dial_y = temp_dial_y
-		else busy = 0
+	user.visible_message("<span class='notice'>[user] starts adjusting [src]'s firing angle and distance.</span>",
+	"<span class='notice'>You begin to adjust [src]'s firing angle and distance to match the new coordinates.</span>")
+
+	playsound(loc, 'sound/items/ratchet.ogg', 25, 1)
```

Comment:
Sure you dont want to move this into ui_act, or at least add a second message?

Reply:
moving it to ui_act would lead to spam

Reply:
I could maybe